g2p/tests/test_studio.py

```diff
@@ -163,83 +169,111 @@ async def test_langs(self):
                         output_text = ""
 
                         # Select the input language
-                        await in_lang_selector.select_option(value=test[0])
+                        # descendants_promise = page.wait
+                        async with page.expect_request(
+                            "**/descendants/" + test_in_lang
+                        ) as req_info:
+                            await in_lang_selector.select_option(value=test_in_lang)
+                        await req_info.value
                         # wait up to max_action_delay ms for input lang to be set
                         # and for mappings to be populated
                         loop_time = 0
```

Comment:
I think this (line 169-228) could be achieved with `await expect(table_status_selector).to_have_attribute("state","loaded")`  you have to import `from playwright.async_api import expect`.
When I tried it test time went from `Ran 4 tests in 86.393s` to `Ran 4 tests in 69.550s` on my local machine.

`to_have_attribute` has a default timeout of 5000ms see https://playwright.dev/python/docs/api/class-locatorassertions#locator-assertions-to-have-attribute

Reply:
Wow, I thought there might be ways to shorten this code, but not this dramatically, and I certainly didn't expect it would speed it up so much.